server/src/main/java/io/crate/statistics/ReservoirSampler.java

```diff
@@ -258,6 +282,8 @@ void setShard(LuceneCollectorExpression<?> collector) {
         @SuppressWarnings("unchecked")
         void collect(int docId) {
             expression.setNextDocId(docId);
+            T value = (T) expression.value();
+            ramAccounting.addBytes(dataType.valueBytes(value));
```

Comment:
Can you double check that the cast is safe without going through dataType.sanitizeValue? E.g. if the source lookup interpretes a `LONG` as `INTEGER` due to the json parser behavior?

Reply:
Do you mean replace the case with `T value = dataType.sanitizeValue(expression.value());`?

Reply:
That would be the safe option but I'm not sure if it is required. (A case where you use a LONG type with a small integer should tell you)

Reply:
Gotcha.

I think this should in fact always work because practically speaking the expression and the sketch will always be parametrized with the same type, but because of the way things are constructed we lose the type information of the expression somewhere along the way.  For now I'll add in the sanitizeValue call.  We may be able to remove it later.